lib/cpp/src/thrift/concurrency/ThreadManager.cpp

```diff
@@ -421,7 +416,7 @@ void ThreadManager::Impl::removeWorker(size_t value) {
   {
     Synchronized s(workerMonitor_);
 
-    while (workerCount_ != workerMaxCount_) {
+    while (workerCount_ > goalCount) {
```

Comment:
This routine blocks until there are fewer actual workers than the number of workers when it was called less the value passed in.  On a busy system where workers are being added this means it might be in here for a while.  I did not change this behavior; simply I ensured that the effects of multiple threads calling removeWorker() at the same time don't clobber each-other's desired goal of how many to remove.

It would be better to have two calls; one to trim the maximum number of workers but it doesn't block; another being a barrier you can use to get to that number.  Callers may want to trim without blocking, for example.